### PR TITLE
Improve setup

### PR DESCRIPTION
**Pull request by typemytype**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -4,27 +4,11 @@
...
-try:
-    import defconAppKit
-except ImportError:
-    print("*** Warning: drawBot requires defconAppKit, see:")
-    print("    https://github.com/typesupply/defconAppKit")
+__version__ = "undefined"
+# read the version number for the settings file
+with open('drawBot/drawBotSettings.py', "r") as settings:
+    source = settings.read()
+    exec(source)
```
[the remark this answers is not in the document]
Comment (typemytype) [reply setup.py]: changed

File: setup.py [hunk 2]
```diff
@@ -4,27 +4,11 @@
...
-    print("*** Warning: drawBot requires vanilla, see:")
-    print("    https://github.com/typesupply/vanilla")
-
-
-try:
-    import defconAppKit
-except ImportError:
-    print("*** Warning: drawBot requires defconAppKit, see:")
-    print("    https://github.com/typesupply/defconAppKit")
+__version__ = "undefined"
```
[the remark this answers is not in the document]
Comment (typemytype) [reply setup.py]: changed

File: setup.py [hunk 3]
```diff
@@ -2,29 +2,17 @@
...
-    import defconAppKit
-except ImportError:
-    print("*** Warning: drawBot requires defconAppKit, see:")
-    print("    https://github.com/typesupply/defconAppKit")
+_versionRE = re.compile(r'__version__\s*=\s*\"([^\"]+)\"')
+# read the version number for the settings file
+with open('drawBot/drawBotSettings.py', "r") as settings:
+    code = settings.read()
+    found = _versionRE.search(code)
+    if found:
```
Comment (justvanrossum) [inline setup.py]: It not being found should be an error: `assert found is not None, "__version__ not found"`

File: setup.py [hunk 4]
```diff
@@ -13,6 +13,8 @@
     found = _versionRE.search(code)
     if found:
         __version__ = found.group(1)
+    else:
```
Comment (justvanrossum) [inline setup.py]: If you do the assert before you get the value, the whole if/else gan go.

**Pull request merged**